stl/inc/chrono

```diff
@@ -2086,6 +2091,720 @@ namespace chrono {
+        } else {
+            seconds _Prev_elapsed{0};
+            for (_It = _Ls_vector.begin(); _It != _Ls_vector.end(); ++_It) {
+                // Positive leap seconds are counted one second earlier, when insertion begins.
+                if (utc_seconds{_It->date().time_since_epoch()} + (_It->_Positive() ? _Prev_elapsed : _It->_Elapsed())
+                    > _Time_floor) {
+                    break;
+                }
+                _Prev_elapsed = _It->_Elapsed();
+            }
+        }
+
+        if (_It != _Ls_vector.begin()) {
+            // Convert to the last leap second before or equal to _Time.
+            const auto& _Last_leap = *--_It;
+            const utc_seconds _Utc_leap_second{_Last_leap.date().time_since_epoch() + _It->_Elapsed() - seconds{1}};
+#ifdef __cpp_lib_concepts
+            const auto _Leap_cmp = _Utc_leap_second <=> _Time_floor;
+#else // ^^^ __cpp_lib_concepts / TRANSITION, GH-395 workaround vvv
+            const auto _Leap_cmp = _Utc_leap_second > _Time_floor  ? strong_ordering::greater
+                                 : _Utc_leap_second == _Time_floor ? strong_ordering::equal
+                                                                   : strong_ordering::less;
+#endif // ^^^ workaround
+            if (_Tzdb._All_ls_positive && _STD is_gt(_Leap_cmp)) {
+                --_It;
+            }
+            return {_Last_leap._Positive() && _STD is_eq(_Leap_cmp), _It->_Elapsed()};
+        } else {
+            return {false, seconds{0}};
+        }
+    }
+
+    class utc_clock {
+    public:
+        using rep                       = system_clock::rep;
+        using period                    = system_clock::period;
+        using duration                  = duration<rep, period>;
+        using time_point                = time_point<utc_clock>;
+        static constexpr bool is_steady = system_clock::is_steady;
+
+        _NODISCARD static time_point now() {
+            return from_sys(system_clock::now());
+        }
+
+        template <class _Duration>
+        _NODISCARD static sys_time<common_type_t<_Duration, seconds>> to_sys(const utc_time<_Duration>& _Utc_time) {
+            using _CT = common_type_t<_Duration, seconds>;
+            const auto _Lsi{get_leap_second_info(_Utc_time)};
+            _CT _Ticks;
+            if (_Lsi.is_leap_second) {
+                const auto _Leap_sec_minus_one = _CHRONO floor<seconds>(_Utc_time.time_since_epoch()) - _Lsi.elapsed;
+                if constexpr (is_integral_v<typename _Duration::rep>) {
+                    constexpr auto _Delta{seconds{1} - _CT{1}};
+                    _Ticks = _Leap_sec_minus_one + _Delta;
+                } else {
+                    const auto _Leap_sec_begin = _CHRONO ceil<_CT>(_Leap_sec_minus_one + seconds{1});
+                    _Ticks                     = _CT{_STD nextafter(_Leap_sec_begin.count(), typename _CT::rep{0})};
+                }

```

Comment:
`duration::rep` could be a custom arithmetic-like type. `<chrono>` algorithms are expected to determine whether `rep` should behave like an integral type or a floating-point type by [`treat_as_floating_point_v<rep>`](https://en.cppreference.com/w/cpp/chrono/treat_as_floating_point).

Reply:
Unfortunately, I'm running out of bandwidth to try to figure out what would be the least wrong thing here. My current best idea is to restrict the `nextafter` branch to `is_floating_point_v` types and for other non-integral types return `_CHRONO floor<_CT>(_Leap_sec_minus_one)`. I don't have any ideas about getting a closer result without risking rounding up to `_Leap_sec_minus_one + 1s` or greater.

Reply:
Thanks @statementreply for noticing this subtle conformance issue. I think it's fine not to handle custom arithmetic types, even for the final merge of `feature/chrono` into `main` - @mnatsuhara is noting this with a card in the Chrono Project, and we'll file a tracking issue if necessary (if it's not resolved before the final merge). Basically I think that no users will encounter this in practice, so it's okay to defer - especially because it'll just be a compiler error.